apps/gallery/js/ImageEditor.js

```diff
@@ -776,10 +780,8 @@ function ImageEditor(imageBlob, container, edits, ready, croponly) {
     this.original.onload = function() {
       // Initialize the crop region to the full size of the original image
       self.resetCropRegion();
-      self.resetPreview();
-
       // Display an edited preview of it
-      self.edit(function() {
+      self.requestNewPreview(function() {
```

Comment:
Okay, so at this point it looks to me like requestNewPreview() must be calling edit() internally. That's not what I'd expect. Since edit() calls generateNewPreview() as needed, I'd expect that requestNewPreview would just set a flag saying that the current preview is out of date. Then, next time edit() is called, the preview will be created before being displayed. If you do it that way, then you'd just call requestNewPreview() before calling edit(), but wouldn't replace your edit() calls.  Of course, I haven't read all of this code yet.